src/gax/src/retry_policy.rs

```diff
@@ -205,6 +205,45 @@ pub trait RetryPolicyExt: RetryPolicy + Sized {
     fn with_attempt_limit(self, maximum_attempts: u32) -> LimitedAttemptCount<Self> {
         LimitedAttemptCount::custom(self, maximum_attempts)
     }
+
+    /// Decorate a [RetryPolicy] to treat all requests as idempotent.
+    ///
+    /// This may be useful is some services where request that are not
+    /// idempotent in general, can be idempotent under some configurations.
+    /// For example, in Google Cloud Storage uploading a new object is not
```

Comment:
Alternatively, we could add `idempotency` to the `storage::RequestOptions`. It already exists for `gax::options::RequestOptions`.

Why do you prefer this?